addons/Media_Recorder/src_webpack/view/button/DownloadButton.jsm

```diff
@@ -3,17 +3,38 @@ import {BlobService} from "../../state/service/BlobService.jsm";
 
 export class DownloadButton extends Button {
 
-    constructor({$view, addonState}) {
+    constructor({$view, addonState, playerController, isMlibro}) {
         super($view);
         this.addonState = addonState;
+        this.playerController = playerController;
+        this.mlibroDownloadUrl = null;
+        this.isMlibro = isMlibro;
+        this.isSafari = DevicesUtils.getBrowserVersion().toLowerCase().indexOf("safari") > -1;
     }
 
     _eventHandler() {
         if (this.addonState.recording) {
-            this.downloadRecording();
+            if (this.isSafari && this.isMlibro) {
+                this.downloadRecordingMLibro();
+            } else {
+                this.downloadRecording();
+            }
         }
     }
 
+    downloadRecordingMLibro() {
+        var self = this;
```

Comment:
@maciej-zawlocki-solwit 
Wydaje mi się, że nie ma potrzeby przypisania tutaj self = this. Arrow functions nie tworzą nowego scope tylko dziedziczą go, stąd też powinien być dostęp do chociażby playerControllera poprzez this (scope obiektu DownloadButton). Ale do weryfikacji czy aby na pewno.